### PR TITLE
Update camera poller to only take pictures in sufficient light.

### PR DESCRIPTION
**Pull request by JeetShetty**
Until now I wasn't really thinking about how the camera poller has access to the queue but doesn't use it. Seems bad? Should I try to find a way to refactor the camera poller code?

#118

### CODE REVIEW

File: greenpithumb/camera_manager.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 # Format of filename to write for camera image file (assumes timestamp is in
 # UTC), as YYYY-MM-DDTHH:MMZ (minutes-level precision).
 _FILENAME_FORMAT = '%Y-%m-%dT%H%MZ.jpg'
+LIGHT_THRESHOLD_PCT = 75
```
Comment (mtlynch) [inline greenpithumb/camera_manager.py]: Also, can we add a comment explaining what this is? Also, I think 60% is enough for decent pictures.

File: greenpithumb/poller.py [hunk 2]
```diff
@@ -256,9 +257,23 @@ def _poll_once(self):
...
+            camera_manager: Manages the capture and saving of photos.
+            light_sensor: An interface for reading the light level.
+        """
+        super(_CameraPollWorker, self).__init__(scheduler, record_queue,
+                                                camera_manager)
+        self._light_sensor = light_sensor
+
     def _poll_once(self):
         """Captures and stores an image."""
-        self._sensor.save_photo()
```
Comment (mtlynch) [inline greenpithumb/poller.py]: Can we push more of the logic into `CameraManager`? Sort of mirroring the design of `PumpManager` with its moisture threshold and `pump_if_needed`.

**Comment by mtlynch** [comment]
>Until now I wasn't really thinking about how the camera poller has access to the queue but doesn't use it. Seems bad? Should I try to find a way to refactor the camera poller code?

Oh yeah, it's a little messy, but I can't think of a way to fix it without adding more complexity. The way that first comes to mind is to create a deeper class hierarchy like

```python
class _SensorPollWorkerBase(object):
  # Base for all poll workers

class RecordProducingPollWorkerBase(_SensorPollWorkerBase):
  # Base class for workers that produce recorsd

class TemperatureWorker(RecordProducingPollWorkerBase):
  ...
```

But I feel like that's more confusing than what we have now.

Any ideas for different approaches? I think as-is it's fine, but maybe there's a clean way to fix it.

**Comment by JeetShetty** [comment]
>Any ideas for different approaches? I think as-is it's fine, but maybe there's a clean way to fix it.

Yeah nothing good comes to mind right now, but I'll keep thinking about it. In the most recent commit I changed the ```camera_poller_factory``` instantiation to use ```None``` instead of the actual record queue, but I'm not sure if this is a good idea.

**Pull request merged**